<?xml version="1.0"?>
<launch>

  <!-- camera_name doubles as namespace -->
  <arg name="camera_name"  default="oak_left"/>
  <arg name="camera_frame" default="$(arg camera_name)_rgb_camera_optical_frame"/>
  <arg name="image_topic"  default="image_rect"/>
  <arg name="publish_tf"   default="false"/>
  <arg name="tags_file"    default="$(find rgmc_vision)/config/tags_test.yaml"/>
  <arg name="launch_prefix" default=""/>

  <!-- Put everything in the camera’s namespace -->
  <group ns="$(arg camera_name)">

    <!-- load exactly into the node’s namespace -->
    <rosparam command="load"
              file="$(find rgmc_vision)/config/settings.yaml"
              ns="apriltag_ros_continuous_node"/>
    <rosparam command="load"
              file="$(arg tags_file)"
              ns="apriltag_ros_continuous_node"/>

    <!-- this node name must match the ns above -->
    <node pkg="apriltag_ros"
          type="apriltag_ros_continuous_node"
          name="apriltag_ros_continuous_node"
          clear_params="false"
          output="screen"
          launch-prefix="$(arg launch_prefix)">

      <remap from="image_rect"  to="rgb/$(arg image_topic)"/>
      <remap from="camera_info" to="rgb/camera_info"/>

      <param name="camera_frame"                value="$(arg camera_frame)"/>
      <param name="publish_tag_detections_image" value="true"/>
      <param name="publish_tf"                  value="$(arg publish_tf)"/>
    </node>
  </group>

</launch>
